packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts

```diff
@@ -13,36 +13,43 @@ import {
 
 import { SchedulingType } from "@calcom/platform-enums";
 
+import { BookerLayouts_2024_06_14 } from "./booker-layouts.input";
 import type { BookingField_2024_06_14 } from "./booking-fields.input";
 import { ValidateBookingFields_2024_06_14 } from "./booking-fields.input";
-import { BookingLimitsCount_2024_06_14, ValidateBookingLimistsCount } from "./booking-limits-count.input";
-import {
-  BookingLimitsDuration_2024_06_14,
-  ValidateBookingLimistsDuration,
-} from "./booking-limits-duration.input";
+import type { BookingLimitsCount_2024_06_14 } from "./booking-limits-count.input";
+import { ValidateBookingLimistsCount } from "./booking-limits-count.input";
+import type { BookingLimitsDuration_2024_06_14 } from "./booking-limits-duration.input";
+import { ValidateBookingLimistsDuration } from "./booking-limits-duration.input";
 import type { BookingWindow_2024_06_14 } from "./booking-window.input";
 import { ValidateBookingWindow } from "./booking-window.input";
+import type { ConfirmationPolicy_2024_06_14 } from "./confirmation-policy.input";
+import { ValidateConfirmationPolicy } from "./confirmation-policy.input";
+import { Disabled_2024_06_14 } from "./disabled.input";
+import { EventTypeColor_2024_06_14 } from "./event-type-color.input";
 import { ValidateLocations_2024_06_14 } from "./locations.input";
 import type { Location_2024_06_14 } from "./locations.input";
 import { Recurrence_2024_06_14 } from "./recurrence.input";
+import { Seats_2024_06_14 } from "./seats.input";
 
 export const CREATE_EVENT_LENGTH_EXAMPLE = 60;
 export const CREATE_EVENT_TITLE_EXAMPLE = "Learn the secrets of masterchief!";
 export const CREATE_EVENT_DESCRIPTION_EXAMPLE =
   "Discover the culinary wonders of the Argentina by making the best flan ever!";
 
-export class CreateEventTypeInput_2024_06_14 {
+export class BaseCreateEventTypeInput_2024_06_14 {
+  @IsOptional()
   @IsInt()
   @Min(1)
-  @DocsProperty({ example: CREATE_EVENT_LENGTH_EXAMPLE })
-  lengthInMinutes!: number;
+  lengthInMinutes?: number;
 
+  @IsOptional()
   @IsString()
   @DocsProperty({ example: CREATE_EVENT_TITLE_EXAMPLE })
-  title!: string;
+  title?: string;
```

Comment:
> These fields were made optional in the base class because it serves as a foundation for various derived classes with different requirements. In `CreateTeamEventTypeInput_2024_06_14`, the specific input class where these fields are essential, they are overridden to be required.
> 
> This approach allows us to enforce stricter validation rules in derived classes, as loosening restrictions in derived classes is not permissible. Since we’re also extending `UpdateEventTypeInput_2024_06_14` from this base class, where these fields need to remain optional, the base class must accommodate this by making them optional. This strategy provides flexibility across different use cases while ensuring that necessary fields are mandatory where needed.

I am not sure I am a fan of this approach, because if in the the base class `BaseCreateEventTypeInput_2024_06_14` everything is optional and another class extends it, it is very hard to keep track of what is needed and what is not needed. The fact that create team event type input overrides a property is a sign in my opinion that we should not derive other classes from a class where everything is optional, because if a class extends a class we assume that whatever `BaseCreateEventTypeInput_2024_06_14` can do also `CreateTeamEventTypeInput_2024_06_14` can do, but since base class has all optional we can't be sure of that.

I think it is better to have separate classes with explicitly defined properties and not have base class with all optional properties.

Reply:
https://github.com/calcom/cal.com/pull/16314/commits/ca1162463e75814654a63c07b4a081ec0de36d49